Mathlib/NumberTheory/Divisors.lean

```diff
@@ -637,21 +634,13 @@ def divisorsAntidiag : (z : ℤ) → Finset (ℤ × ℤ)
       simp +contextual [s, disjoint_left, eq_comm, forall_swap (α := _ * _ = _)]
 
 @[simp]
-theorem mem_divisors : ∀ {z x}, x ∈ divisors z ↔ x ∣ z ∧ z ≠ 0
-  | (n : ℕ), (m : ℕ) => by
-    simp [divisors]
-    norm_cast
-    simp
-  | (negSucc n), (m : ℕ) => by
-    simp [divisors]
-    norm_cast
-  | (n : ℕ), (negSucc m) => by
-    simp [divisors]
-    norm_cast
-    simp
-  | (negSucc n), (negSucc m) => by
-    simp [divisors]
-    norm_cast
+theorem mem_divisors : ∀ {z x}, x ∈ divisors z ↔ x ∣ z ∧ z ≠ 0 := by
+  rintro z x
+  rw [ne_eq, ← natAbs_eq_zero (a := z), ← natAbs_dvd_natAbs, ← Nat.mem_divisors]
+  simp only [divisors, disjUnion_eq_union, mem_union]
+  match x with
+  | (n : ℕ) => simp
+  | (negSucc n) => simp
```

Comment:
```suggestion
theorem mem_divisors {z x} : x ∈ divisors z ↔ x ∣ z ∧ z ≠ 0 := by
  rw [ne_eq, ← natAbs_eq_zero (a := z), ← natAbs_dvd_natAbs, ← Nat.mem_divisors]
  simp only [divisors, disjUnion_eq_union, mem_union]
  obtain ⟨n, rfl | rfl⟩ := x.eq_nat_or_neg <;> simp
```
1) The convention is to have universally quantified arguments and hypothesis "left of the colon" (this saves the "intro" line in the proof).
2) Relying on the implementation details of ℤ is a bit ugly and can be avoided by using the API.